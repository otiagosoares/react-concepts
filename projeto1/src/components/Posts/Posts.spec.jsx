import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { Posts } from '.';

const props = {
  posts: [
    { id: 1, title: 'title 1', body: 'body', cover: 'img/cover.png' },
    { id: 2, title: 'title 2', body: 'body', cover: 'img/cover.png' },
    { id: 3, title: 'title 3', body: 'body', cover: 'img/cover.png' },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
