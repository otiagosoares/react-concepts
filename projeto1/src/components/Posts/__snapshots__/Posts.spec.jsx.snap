// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <article
    class="post"
  >
    <div
      class="post-thumb"
    >
      <img
        alt="title 1"
        src="img/cover.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title 1
         - 
        1
      </h2>
      <p>
        body
      </p>
    </div>
  </article>
  <article
    class="post"
  >
    <div
      class="post-thumb"
    >
      <img
        alt="title 2"
        src="img/cover.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title 2
         - 
        2
      </h2>
      <p>
        body
      </p>
    </div>
  </article>
  <article
    class="post"
  >
    <div
      class="post-thumb"
    >
      <img
        alt="title 3"
        src="img/cover.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title 3
         - 
        3
      </h2>
      <p>
        body
      </p>
    </div>
  </article>
</div>
`;
